sqlglot/dialects/bigquery.py

```diff
@@ -104,6 +107,33 @@ def _unqualify_unnest(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def _pushdown_cte_column_names(expression: exp.Expression) -> exp.Expression:
+    """BigQuery doesn't allow column names when defining a CTE, so we try to push them down."""
+    if isinstance(expression, exp.CTE) and expression.alias_column_names:
+        cte_query = expression.this
+        column_names = expression.alias_column_names
+
+        if cte_query.is_star:
+            logger.warning(
+                "Can't push down CTE column names for star queries. Run the query through"
+                " the optimizer or use 'qualify' to expand the star projections first."
+            )
+            return expression
+
+        expression.args["alias"].set("columns", None)
```

Comment:
expression.args["alias"].pop("columns")

Reply:
Will this work with `columns` being a list?

Reply:
aren't we just getting rid of columns?

Reply:
Maybe we can update the API of `pop` so that it receives a `key` arg to set to `None`.

Reply:
> aren't we just getting rid of columns?

Yep, it's just that this API is not currently supported. No big issue though, I can just add an optional arg in `pop`.